src/index.ts

```diff
@@ -96,7 +96,8 @@ export default function pluginTurbosnap({
         mod.importedIds
           .concat(mod.dynamicallyImportedIds)
           .filter((name) => isUserCode(name))
-          .forEach((depId) => {
+          .forEach((depIdUnsafe) => {
+            const depId = normalize(depIdUnsafe);
```

Comment:
I see, this seems to be due to the query params on the vue file's id, for example:

```
/Users/ianvs/code/repro-storybook/src/stories/Button.vue?vue&type=script&setup=true&lang.ts
```

By normalizing here, a bit earlier in the process, we can de-duplicate those.  But, we end up with a bit of a strange situation, like:

```json
    {
      "id": "./src/stories/Button.vue",
      "name": "./src/stories/Button.vue",
      "reasons": [
        {
          "moduleName": "./src/stories/Button.stories.ts"
        },
        {
          "moduleName": "./src/stories/Button.vue"
        }
      ]
    },
```

This says that `Button.vue` was included in the build because of `Button.vue`, which is a bit circular.  Maybe we should detect that and filter out any `reasons` which match the `id`.  What do you think?

Reply:
Correct, I forget to mention that it's related to Vue's SFC, which produces query params to match <script setup>.

Good catch with `Button.vue`, I think we should definitely filter that out.

Reply:
Cool.  Would you like to do that, or should I take a shot at it?

Reply:
@IanVS just pushed the changes, thank you :)